src/modules/addRemove.js

```diff
@@ -0,0 +1,197 @@
+
+  taskHolder.forEach((instance) => {
+    const { index, description, completed } = instance;
+
+    const taskListItem = document.createElement('li');
+    taskListItem.classList.add('listInstance');
+    taskListItem.setAttribute('data-index', index);
+    taskListItem.innerHTML = `
+      <input type="checkbox" name="${index}" id="${index}" class='checkbox'>
+      <span class='descriptionText' data-index='${index}'>${description}</span>
+      <img src=${dots} class='threedots'data-index='${index}'>
+      <img src=${trash} class='trash' data-index='${index}'>
+      `;
+    tasklist.append(taskListItem);
+
+    const listStruc = document.querySelectorAll('.listInstance');
+    listStruc.forEach((inst) => {
+      // edit description
+      const descriptionTexts = inst.querySelectorAll('.descriptionText[data-index]');
+      descriptionTexts.forEach((span) => {
+        span.addEventListener('click', () => {
+          const input = document.createElement('input');
+          input.classList.add('input-2');
+          input.type = 'text';
+          input.value = span.innerText;
+
+          input.addEventListener('blur', () => {
+            span.innerText = input.value;
+            instance.description = input.value;
+            setTimeout(() => {
+              input.replaceWith(span);
+            }, 0);
+            localStorage.setItem('taskList', JSON.stringify(taskHolder));
+          });
+
+          input.addEventListener('keydown', (event) => {
+            if (event.key === 'Enter') {
+              span.innerText = input.value;
+              instance.description = input.value;
+              setTimeout(() => {
+                input.replaceWith(span);
+              }, 0);
+            }
+            localStorage.setItem('taskList', JSON.stringify(taskHolder));
+          });
+          span.replaceWith(input);
+          input.focus();
+        });
+        localStorage.setItem('taskList', JSON.stringify(taskHolder));
+      });
+    });
+    bookSave();
+  });
+};
+
+inputField.addEventListener('keydown', (event) => {
+  if (event.key === 'Enter' && !(inputField.value === '')) {
+    taskCreator();
+    addToDom();
+    clearInput();
+  }
+});
+
+arrowHolder.addEventListener('click', () => {
+  if (!(inputField.value === '')) {
+    taskCreator();
+    addToDom();
+    clearInput();
+  }
+});
+
+// render list-section dynamically when page loads
+window.addEventListener('load', () => {
+  addToDom();
+});
+
+// handle clicks
+const handleDeleteClick = (event) => {
+  const listItem = event.target.closest('li');
+  const listItemIndex = parseInt(listItem.dataset.index, 10);
+  const indexInTaskHolder = taskHolder.findIndex((task) => task.index === listItemIndex);
+  if (indexInTaskHolder !== -1) {
+    taskHolder.splice(indexInTaskHolder, 1);
+  } else if (indexInTaskHolder === 0) {
+    taskHolder.splice(0, taskHolder.length);
+  }
+  listItem.remove();
+  localStorage.setItem('taskList', JSON.stringify(taskHolder));
+};
```

Comment:
Good job implementing the delete feature. However, it behaves quite unexpected as per when an item is removed, the index is not updating as described [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should%20update%20all%20remaining%20items%27%20indexes%2C%20so%20they%20represent%20the%20current%20list%20order%20and%20are%20unique(i.e.%20if%20you%27re%20deleting%20the%20first%20task%20index%201%20from%20the%20list%2C%20the%20index%20of%20the%20next%20task(2)%20should%20set%20to%201)..). I think you should resolve this because it making the delete leave behind one item sometimes. Here is a video of this behaviour.


[screen-capture (1).webm](https://github.com/gisachris/To-Do-List/assets/84930355/ef04af54-0871-4fde-b694-ce39c2a8b9f8)